hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -2559,7 +2572,7 @@ public boolean needsCompaction() {
    * @return cache configuration for this Store.
    */
   public CacheConfig getCacheConfig() {
-    return this.cacheConf;
+    return storeContext.getCacheConf();
   }
 
   public static final long FIXED_OVERHEAD = ClassSize.estimateBase(HStore.class, false);
```

Comment:
Does DEEP_OVERHEAD and heapSize() need to change? Does TestHeapSize still pass?

Reply:
TestHeapSize still passed when I checked it locally.

from what I understood from the `FIXED_OVERHEAD` were being calculated by `ClassSize.estimateBase`, it will automatically calculate the our newly added fields and reference included the change of `HStoreContext storeContext`. 

So for the `DEEP_OVERHEAD` that takes new calculated `FIXED_OVERHEAD` to come up the heap size, it should be done already. Or did I miss something here?

Reply:
this has been change as well, can you have another look ?